modules/terraform/plan.go

```diff
@@ -0,0 +1,37 @@
+package terraform
+
+import (
+	"testing"
+)
+
+// InitAndPlan runs terraform init and plan with the given options and return stdout/stderr from the apply command.
+func InitAndPlan(t *testing.T, options *Options) int {
+	exitCode, err := InitAndPlanE(t, options)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return exitCode
+}
+
+// InitAndPlanE runs terraform init and plan with the given options and return stdout/stderr from the apply command.
+func InitAndPlanE(t *testing.T, options *Options) (int, error) {
+	if _, err := InitE(t, options); err != nil {
+		return TerraformPlanExceptionExitCode, err
+	}
+
+	return PlanExitCodeE(t, options)
+}
+
+// PlanExitCode runs terraform apply with the given options and returns the detailed exitcode.
+func PlanExitCode(t *testing.T, options *Options) int {
+	exitCode, err := PlanExitCodeE(t, options)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return exitCode
+}
+
+// PlanExitCodeE runs terraform apply with the given options and returns the detailed exitcode.
+func PlanExitCodeE(t *testing.T, options *Options) (int, error) {
+	return GetExitCodeForTerraformCommandE(t, options, FormatArgs(options.Vars, "plan", "-input=false", "-lock=false", "-detailed-exitcode")...)
```

Comment:
It's still not obvious to me why we would want `-lock=false` here... If people are making changes concurrently to the same state, we want the lock to kick in!

Reply:
If developer A is running a long running update and developer B has pushed commit (in a different module) to the repo which triggered a build on CI, B's commit might fail the build waiting for a lock(exit 1) whereas if there is deviation the plan should only detect change(exit 2). do not want to put in an arbitrary value for the timeout and hence want to go ahead with non locking. thoughts?

Reply:
Isn't this code just used in automated tests? What other devs are operating on this state concurrently?

Reply:
my bad. got confused in my thoughts. fixed.